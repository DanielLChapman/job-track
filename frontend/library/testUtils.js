import casual from 'casual';

casual.seed(231313);

const fakeUser = (overrides) => ({
    __typename: 'User',
    id: '4234',
    name: casual.name,
    jobs: [],
    email: casual.email,
    ...overrides,
  });

  const fakeJob = (overrides) => ({
    __typename: 'User',
    id: '4234',
    name: casual.name,
    publishedAt: casual.date('YYYY-MM-DD'),
    applicationDate: casual.date('YYYY-MM-DD'),
    status: "waiting",
    author: fakeUser().id,
    salaryExpectation: 0,
    notes: casual.sentence,
    ...overrides,
  });

  export {
    fakeUser,
    fakeJob
  }
