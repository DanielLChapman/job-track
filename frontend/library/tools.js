export const convertNum = (num) => {
    let t = num.toString();
    if (t.length < 3) {
        t = "$" + t
        return t;
    }
    let result = [];
    let count = 1;
    for (let i = t.length - 1; i >= 0; i--) {
        result.unshift(t[i]);
        
        if (count % 3 === 0 && i !== 0) {
            result.unshift(',')
        }
        count++;
    }

    result.unshift('$')

    return result.join('');

}