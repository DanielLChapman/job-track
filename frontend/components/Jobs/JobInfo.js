import React from 'react';
import PropTypes from 'prop-types';
import { convertNum } from '../../library/tools';

JobInfo.propTypes = {
    job: PropTypes.object.isRequired,
};

function JobInfo({job}) {
    return (
        <ul>
            <li><h2>{job.name}</h2></li>
            <li><h5>{job.status}</h5></li>
            <li><h5>{convertNum(job.salaryExpectation)}</h5></li>
            <li><h5>{job.notes}</h5></li>
        </ul>
    );
}

export default JobInfo;