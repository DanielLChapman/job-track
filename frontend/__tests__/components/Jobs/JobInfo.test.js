import {act, screen, render} from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import wait from 'waait';
import { fakeJob } from '../../../library/testUtils';
import JobInfo from '../../../components/Jobs/JobInfo';

const job = fakeJob();

describe('It insures the correct information is displayed', () => {
    it ('renders correctly when a job is passed in', async () => {

        const { container, debug} = render(
            <MockedProvider>
                <JobInfo job={job} />
            </MockedProvider>
        )

        await act(async() => {
            await wait(0);
        });

        await screen.findByText(job.name);

        expect(container).toMatchSnapshot();
    })
})