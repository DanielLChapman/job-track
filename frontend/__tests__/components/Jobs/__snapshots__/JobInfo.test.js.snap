// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It insures the correct information is displayed renders correctly when a job is passed in 1`] = `
<div>
  <ul>
    <li>
      <h2>
        Mrs. Constantin Dickinson
      </h2>
    </li>
    <li>
      <h5>
        waiting
      </h5>
    </li>
    <li>
      <h5>
        $0
      </h5>
    </li>
    <li>
      <h5>
        Neque quia vero sint ab dolorem deserunt provident.
      </h5>
    </li>
  </ul>
</div>
`;
